src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1525,26 +1525,32 @@ def handle_merge(existing, addition, key, replace):
         existing[key] = addition[key]
         return
 
-    for i in addition[key]:
-        for j in existing[key]:
-            if i['name'] == j['name']:
-                if replace or i == j:
-                    existing[key].remove(j)
+    i = addition[key][0]
+    temp_list = []
+    for j in existing[key]:
+        remove_flag = False
+        if i['name'] == j['name']:
```

Comment:
No loop for `addition[key]`? It seems only `addition[key][0]` is checked.

Reply:
Providing some explanation here on why I removed iteration over addition list: handle_merge() function is called by merge_kubernetes_configurations() function in the code. The latter assumes that addition list has only one object. This can be confirmed from following lines in that function:

```
if context_name is not None:
        addition['contexts'][0]['name'] = context_name
        addition['contexts'][0]['context']['cluster'] = context_name
        addition['clusters'][0]['name'] = context_name
        addition['current-context'] = context_name
```

Hence, removing the outer loop to just use the first object made sense to me.